crates/forge/src/running.rs

```diff
@@ -229,35 +234,42 @@ pub(crate) fn run_test_case(
         &string_to_hint,
     );
 
-    Ok(runner.run_function(
+    match runner.run_function(
         runner.find_function(case.name.as_str())?,
         &mut test_execution_syscall_handler,
         hints_dict,
         instructions,
         builtins,
-    )?)
+    ) {
+        Ok(result) => Ok(TestCaseSummary::from_run_result(result, case, args)),
+        // CairoRunError comes from VirtualMachineError which may come from HintException that originates in the TestSyscallExecutionHandler
+        Err(RunnerError::CairoRunError(error)) => Ok(TestCaseSummary::Failed {
+            name: case.name.clone(),
+            msg: Some(format!(
+                "\n    {}\n",
+                error.to_string().replace(" Custom Hint Error: ", "\n    ")
+            )),
+            arguments: args,
+            fuzzing_statistic: None,
+        }),
+        Err(err) => Err(err.into()),
```

Comment:
Those are other variants of [`RunnerError`](https://github.com/starkware-libs/cairo/blob/main/crates/cairo-lang-runner/src/lib.rs#L47), I don't think they should ever happen but I'm not sure. I decided to bubble an error, it kills the whole runner anyways but with a nice error message